owltorrent-common/src/main/java/edu/rice/owltorrent/common/entity/Peer.java

```diff
@@ -20,6 +21,8 @@
 
   @Getter private final Torrent torrent;
 
+  @Getter @Setter private Bitfield bitfield = Bitfield.EMPTY_BITFIELD_SINGLETON;
```

Comment:
Have the setter to check and throw an exception when the setter is called and the current bitfield is not the empty singleton. (A peer should only ever send bitfield as the first message and only once. )